WordPress/src/jetpack/java/org/wordpress/android/ui/accounts/login/LoginPrologueRevampedFragment.kt

```diff
@@ -64,6 +77,35 @@ class LoginPrologueRevampedFragment : Fragment() {
     }
 }
 
+/**
+ * This composable launches an effect to continuously update the view model by providing the elapsed
+ * time between frames. Velocity and position are recalculated for each frame, with the resulting
+ * position provided here to be consumed by nested children composables.
+ */
+@Composable
+private fun PositionProvider(
+    viewModel: LoginPrologueRevampedViewModel = viewModel(),
+    content: @Composable () -> Unit
+) {
+    val position = viewModel.positionData.observeAsState(0f)
+    CompositionLocalProvider(LocalPosition provides position.value) {
+        LaunchedEffect(Unit) {
+            var lastFrameNanos: Long? = null
+            while (isActive) {
+                val currentFrameNanos = awaitFrame()
```

Comment:
@mkevins I've found this issue in the Android Public Tracker:
[Add a lint check for using awaitFrame inside Compose code 
](https://issuetracker.google.com/issues/229748038) Saying we shouldn't use `awaitFrame` (hint: I was researching `awaitFrame`) in Compose code, but rather the `withFrameNanos` from [MonotonicFrameClock](https://developer.android.com/reference/kotlin/androidx/compose/runtime/MonotonicFrameClock#withFrameNanos(kotlin.Function1)) which is Compose's own time source.

I will adapt the code to use that instead and test if it still works as expected

Reply:
Update: Still works nicely, I applied the change in https://github.com/wordpress-mobile/WordPress-Android/pull/17207/commits/20e21a4b3964054f2898e56e0833cdf53249a6b1

Reply:
Oh nice find Ovi! This is a good improvement that will helpful for supporting possible future UI testing 👍